File: Day-05/solution.py
sentence = input('Enter a sentence: ')

if not sentence.strip():
    print("Empty sentence!")
    exit()

char_list = list(sentence)
characters = [char for char in char_list if char.strip()]
char_freq = {}
char_map = {}

for char in characters:
    lower_char = char.lower()
    char_freq[lower_char] = char_freq.get(lower_char, 0) + 1

    if lower_char not in char_map:
        char_map[lower_char] = char

sorted_chars = sorted(char_freq.items(), key=lambda item: item[1], reverse=True)

print('Character frequencies:')

for lower_char, freq in sorted_chars:
    display_char = char_map[lower_char]

    print(f"{display_char} => {freq}")

File: Day-05/solution.js
import readline from 'readline';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter a sentence: ', (sentence) => {
  if (sentence.trim().length === 0) {
    console.log("Empty sentence!");
    rl.close();

    return;
  }

  const charArr = sentence.split('');
  const characters = charArr.filter((char) => char !== ' ');
  const charFreq = {};
  const charMap = {};

  characters.forEach((char) => {
    let lowerChar = char.toLowerCase();
    charFreq[lowerChar] = (charFreq[lowerChar] || 0) + 1;

    if (!charMap[lowerChar]) {
      charMap[lowerChar] = char;
    }
  });

  const sorted = Object.entries(charFreq).sort((a, b) => b[1] - a[1]);

  console.log('Character frequencies:');

  sorted.forEach(([lowerChar, freq]) => {
    const displayChar = charMap[lowerChar];

    console.log(`${displayChar} => ${freq}`);
  });

  rl.close();
});
